conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/processing/BlocksProcessing.scala

```diff
@@ -0,0 +1,80 @@
+package tech.cryptonomic.conseil.indexer.tezos.processing
+
+import com.typesafe.scalalogging.LazyLogging
+import scala.util.{Failure, Success, Try}
+import scala.concurrent.{ExecutionContext, Future}
+import slick.jdbc.PostgresProfile.api._
+
+import tech.cryptonomic.conseil.indexer.tezos.{
+  TezosNamesOperations,
+  TezosNodeOperator,
+  TezosGovernanceOperations,
+  TezosDatabaseOperations => TezosDb
+}
+import tech.cryptonomic.conseil.common.tezos.TezosTypes.{Block, Voting}
+import tech.cryptonomic.conseil.common.tezos.TezosTypes.Syntax._
+import tech.cryptonomic.conseil.indexer.tezos.michelson.contracts.TokenContracts
+import tech.cryptonomic.conseil.indexer.tezos.michelson.contracts.TNSContract
+
+class BlocksProcessing(
+    nodeOperator: TezosNodeOperator,
+    db: Database,
+    tnsOperations: TezosNamesOperations,
+    accountsProcessing: AccountsProcessing,
+    bakersProcessing: BakersProcessing
+)(implicit tokens: TokenContracts, tns: TNSContract)
+    extends LazyLogging {
+
+  /* will store a single page of block results */
+  private[tezos] def processBlocksPage(results: nodeOperator.BlockFetchingResults)(
+      implicit ec: ExecutionContext
+  ): Future[Int] = {
+    def logBlockOutcome[A]: PartialFunction[Try[Option[A]], Unit] = {
+      case Success(accountsCount) =>
+        logger.info(
+          "Wrote {} blocks to the database, checkpoint stored for{} account updates",
+          results.size,
+          accountsCount.fold("")(" " + _)
```

Comment:
`accountsCount` suggests we are talking about `Numeric`, but we are folding them to a bit different format. What would be an example outcome from this expression? Not sure how it would fit to the logging message.

Reply:
It's an optional number, and I need a result that will be empty as a string when the value is not defined.
Folding provides conversion of the number to a string (with the leading space), with a default of empty string.
See the previous comment for the example output

Reply:
Yea, related to above. Now I understand that.